### PR TITLE
setup testing library

### PR DESCRIPTION
**Pull request by JuAnmdq**
## Description
Setup testing-library for integration tests

- Setup testing-library
- Remove unused enzyme
- Extract jest configuration in _jest.config.js_ file, in order to reduce package.json. 

## References
- [Testing library - Automated tests with built-in expect assertions](https://testing-library.com)

### CODE REVIEW

File: jest.config.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+module.exports = {
+  moduleNameMapper: {
+    '\\.(css|scss)$': 'identity-obj-proxy',
+    '^.+\\.(jpg|jpeg|png|gif|eot|otf|webp|svg|ttf|woff|woff2|mp4|webm|wav|mp3|m4a|aac|oga|ico)$':
+      '<rootDir>/tools/fileMock.js'
```
Comment (nsantos16) [inline jest.config.js:5]: I think you have to add a module mapper for svg files that uses `svgr`, check this [doc](https://react-svgr.com/docs/jest/)

File: package.json [hunk 2]
```diff
@@ -114,6 +117,10 @@
     "@cypress/code-coverage": "1.8.0",
     "@cypress/webpack-preprocessor": "4.0.3",
     "@svgr/webpack": "4.3.2",
+    "@testing-library/dom": "^7.29.4",
```
Comment (nsantos16) [inline package.json:120]: Maybe we can add too `eslint-plugin-jest-dom` and `eslint-plugin-testing-library` to enhancement the linter around the testing files

Comment (nsantos16) [reply package.json:120]: Also, WDYT about `jest-axe` and `msw`? Do you have experience with these libs?

**Review by nsantos16 — APPROVED** [review]

File: src/components/common/Header/index.test.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+test('Header renders without crashing', () => {
+  const { asFragment } = render(<Header />);
+  expect(asFragment).toBeDefined();
+});
+
+test('Header shows the expected header title', () => {
+  render(<Header />);
+  expect(screen.getByRole('heading', { level: 2, name: /header.../i })).toBeInTheDocument();
+});
```
Comment (nsantos16) [inline src/components/common/Header/index.test.js:14]: Maybe wrap these two tests into a `describe` method and render `Header` one time? 🤔

Comment (JuAnmdq) [reply src/components/common/Header/index.test.js:14]: Following for Kent. C. Dodds's code `test` looks a lot cleaner specially when there are many independent tests,  describe('') and when('') can be useful when need before hook to setup or stub for special cases.

Although makes sense, to only use one render when shows a default prop component + .toBeDefined(), in this case we can wrap in one test.

[Integration test - Kent. C. Dodds](https://kentcdodds.com/blog/unit-vs-integration-vs-e2e-tests)

File: package.json (hunk 2 above)
Comment (JuAnmdq) [reply package.json:120]: Yes, you're right about `eslint-plugin-jest-dom` and `eslint-plugin-testing-library`, we can add this dev dependencies.
I never hear about `jest-axe` and `msw` libs.

File: jest.config.js (hunk 1 above)
Comment (JuAnmdq) [reply jest.config.js:6]: Done thanks, I added this module mapper exclusive for svgs. I added into `tools` folder in order to get all utilities there.

File: package.json (hunk 2 above)
Comment (JuAnmdq) [reply package.json:120]: Done with `eslint-plugin-jest-dom` and `eslint-plugin-testing-library` added with recommended settings for each one

Comment (nsantos16) [reply package.json:120]: Awesome 👏

File: src/components/common/Header/index.test.js (hunk 3 above)
Comment (JuAnmdq) [reply src/components/common/Header/index.test.js]: As we talk, I will render once in a test's set

File: package.json (hunk 2 above)
Comment (JuAnmdq) [reply package.json:120]: I will research / implement jest-axe and msw in next `PRs` as a technical debt

**Pull request merged**